### PR TITLE
Updated problem_06 in Project Euler

### PR DESCRIPTION
**Pull request by shellhub**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by spamegg1** [comment]
In `sol1.py` it might be better to use `sum_of_squares` and `sum_of_ints` instead of `sum_a` and `sum_b`.

In `sol2.py` it might be better to do (using integer division `//` to avoid float):
```python
sum_cubes = (n * (n + 1) // 2) ** 2
sum_squares = n * (n + 1) * (2 * n + 1) // 6
return sum_cubes - sum_squares
```

**Comment by shellhub** [comment]
> In `sol1.py` it might be better to use `sum_of_squares` and `sum_of_ints` instead of `sum_a` and `sum_b`.
> 
> In `sol2.py` it might be better to do (using integer division `//` to avoid float):
> 
> ```python
> sum_cubes = (n * (n + 1) // 2) ** 2
> sum_squares = n * (n + 1) * (2 * n + 1) // 6
> return sum_cubes - sum_squares
> ```

I agree. And you ? @cclauss

**Comment by cclauss** [comment]
We should test the solutions against each other.  Please create our more file called `test_solutions.py` that looks something like this:
```python
from sol1 import solution as sol1
from sol2 import solution as sol2
from sol3 import solution as sol3
from sol4 import solution as sol4


def test_solutions()
    """
    >>> test_solutions()
    """
    assert sol1() == sol2() == sol3() == sol4()


if __name__ == "__main__":
    test_solutions()
```

File: project_euler/problem_06/test_solutions.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    >>> test_solutions()
+    """
+    assert sol1(10) == sol2(10) == sol3(10) == sol4(10) == 2640
+    assert sol1(15) == sol2(15) == sol3(15) == sol4(15) == 13160
+    assert sol1(20) == sol2(20) == sol3(20) == sol4(20) == 41230
+    assert sol1(50) == sol2(50) == sol3(50) == sol4(50) == 1582700
+
+
+if __name__ == '__main__':
+    test_solutions()
```
Comment (cclauss) [inline project_euler/problem_06/test_solutions.py]: Python files end with one and only one \n.

**Comment by shellhub** [comment]
@cclauss build passed. Thanks :)

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
Is there a website that provides all the answers to the Euler puzzles so we could test the solutions against those answers?

**Comment by shellhub** [comment]
I have no results after google.😄

**Comment by spamegg1** [comment]
> Is there a website that provides all the answers to the Euler puzzles so we could test the solutions against those answers?

https://github.com/nayuki/Project-Euler-solutions

**Comment by cclauss** [comment]
```python
import requests
url = "https://raw.githubusercontent.com/nayuki/Project-Euler-solutions/master/Answers.txt"
lines = [line.replace(":", "").split()[1:] for line in requests.get(url).text.splitlines() if line.startswith("Problem")]
answers = {int(key): value for key, value in lines}
```

**Comment by nayuki** [comment]
https://github.com/luckytoilet/projecteuler-solutions/blob/master/Solutions.md

**Comment by cclauss** [comment]
@dhruvmanila @nayuki @shellhub @spamegg1 The Euler tests [really slow down out automated testing](https://travis-ci.com/github/TheAlgorithms/Python/builds/184870536#L1421) so I was thinking about creating a second Travis CI job that would run in parallel just to test the Euler code in parallel with the main Travis CI job testing all other algorithms.

* The main Travis CI job continues to run just like today but it skips over the `project_euler` directory.
* In parallel, the new second Travis CI job only tests the `project_euler` directory.
    * It does not need to install all the dependencies so it can start testing faster.
    * A new test can import and run each Euler solution to ensure that it produces the same answer as the lists above.
    * We will have to separate lists of 10 slowest tests to expose more opportunities for optimization.

Is anyone interested to work on this with me?

**Comment by dhruvmanila** [comment]
Yes, I will help. I will take a look at how to implement parallel jobs in Travis CI.

- Ignore the `project_euler` directory from `pytest`
- Use matrix to configure Travis to do parallel jobs
- Check for dependencies in the `project_euler` folder if there are any
- Implement a new test to check for each Euler answer with the given list

One question: Does every Euler solution contains the implementation which prints the actual answer? If it's a maybe then we will have to check for that as well.

**Comment by cclauss** [comment]
> Use matrix to configure Travis to do parallel jobs

`matrix` is deprecated in Travis CI so use `jobs` instead.

> Does every Euler solution contains the implementation which prints the actual answer?

According to CONTRIBUTING.md, algorithmic functions should not `print()` so let’s just test the algorithmic functions.

**Comment by cclauss** [comment]
We should raise issues on the files that do not contain pure algorithmic functions that do not print.  These will be good issues for Hacktoberfest!

**Comment by dhruvmanila** [comment]
Alright, I did it with using `jobs`.
Take a look at it: https://travis-ci.org/github/dhruvmanila/Python/builds/729544804

Also, `mypy` is showing errors in some files within the project_euler directory.

**Pull request merged**